tsinfer/formats.py

```diff
@@ -247,8 +259,15 @@ def add(self, **kwargs):
             self._queue_flush_buffer()
         offset = self.num_buffered_items[self.write_buffer]
         for key, value in kwargs.items():
-            if "full_haplotype" in key:
-                self.buffers[key][self.write_buffer][:, offset, 0] = value
+            # Here we have to special case the haplotype for performance
+            # reasons, as writing the full haplotype is expensive.
+            if key == "haplotype":
```

Comment:
Could make this a bit more readable by pulling start and end out into local vars (rather then looking up kwargs repeatedly)

Reply:
Fixed